code/scripting/api/libs/mission.cpp

```diff
@@ -83,10 +83,40 @@ extern TIMESTAMP Game_shudder_time;
 extern int Game_shudder_total;
 extern float Game_shudder_intensity;
 
+constexpr int COUNT_OBJECTS = -1000;
+
+// Returns the indexth object subclass, or counts the number of indexable objects
+template <typename A>
+int object_subclass_at_index(A object_subclass_array, size_t array_size, int index = COUNT_OBJECTS)
```

Comment:
It's much too confusing to have two completely different kinds of interactions handled by the same function, even if their functionality would be largely similar, and the fact that you have to explicitly preface when you call this function to count stuff with `this function call counts the X that we can access` shows that its too confusing otherwise at a glance. I would definitely ask for a separate function `count_objects()` or something with a clearer name, that does nothing but call `object_subclass_at_index()` with `COUNT_OBJECTS` and to make its last parameter mandatory.

Reply:
Ok, fair point